File: apps/api/app/schemas/feedback.py
from pydantic import BaseModel, Field


ALLOWED_FEEDBACK_TAGS = (
    "report_useful",
    "waited_too_long",
    "bugs",
    "easy_to_use",
    "confusing",
    "would_recommend",
)


class FeedbackStatusOut(BaseModel):
    submitted: bool


class FeedbackCreate(BaseModel):
    project_id: str | None = None
    tags: list[str] = Field(default_factory=list, max_length=8)
    message: str = Field(default="", max_length=2000)


class FeedbackOut(BaseModel):
    id: str
    submitted: bool = True

File: apps/web/src/lib/api/feedback.ts
import { apiFetch } from '@/lib/api/client'

export type FeedbackTag =
  | 'report_useful'
  | 'waited_too_long'
  | 'bugs'
  | 'easy_to_use'
  | 'confusing'
  | 'would_recommend'

export function getFeedbackStatus() {
  return apiFetch<{ submitted: boolean }>('/api/v1/feedback/status')
}

export function submitFeedback(payload: {
  project_id?: string
  tags: FeedbackTag[]
  message?: string
}) {
  return apiFetch<{ id: string; submitted: boolean }>('/api/v1/feedback', {
    method: 'POST',
    body: JSON.stringify(payload),
  })
}
